ui-manchette/src/components/Manchette.tsx

```diff
@@ -1,51 +1,202 @@
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
+import cx from 'classnames';
 import OperationalPointList from './OperationalPointList';
-import { OperationalPointType } from '../types';
+import type {
+  OperationalPointType,
+  ProjectPathTrainResult,
+  StyledOperationalPointType,
+} from '../types';
 import { ZoomIn, ZoomOut } from '@osrd-project/ui-icons';
-
+import { SpaceTimeChart, PathLayer } from '@osrd-project/ui-spacetimechart';
+import {
+  calcOperationalPointsToDisplay,
+  getOperationalPointsWithPosition,
+  getScales,
+  operationalPointsHeight,
+} from './helpers';
+import type { OperationalPoint, SpaceScale } from '@osrd-project/ui-spacetimechart/dist/lib/types';
+import { INITIAL_OP_LIST_HEIGHT, INITIAL_SPACE_TIME_CHART_HEIGHT } from './consts';
+import { getDiff } from '../utils/vector';
+import { useIsOverflow } from '../hooks/useIsOverFlow';
 type ManchetteProps = {
   operationalPoints: OperationalPointType[];
-  children?: React.ReactNode;
-  isProportionnal?: boolean;
+  projectPathTrainResult: ProjectPathTrainResult[];
 };
+import usePaths from '../hooks/usePaths';
+
+const Manchette: React.FC<ManchetteProps> = ({ operationalPoints, projectPathTrainResult }) => {
+  const manchette = useRef<HTMLDivElement>(null);
+
+  const [zoomY, setZoomY] = useState(1);
```

Comment:
Why do you use so many separate `setState` instead of one single "larger" `setState` such as something like this for instance?

```typescript
const [state, setState] = useState({
  zoomX: 1,
  zoomY: 1,
  offsetX: 0,
  offsetY: 0,
  // ...
});
```

The "more compact" version seems easier to read for me (it's easy to have a full overview of what's in the state). Do you see any pros on your "split" version?